mautic/mautic
# Enable PHPStan Extensions

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

This PR introduce PHPStan extensions that allows a better scanning to the code base and stop any technical debt in future PRs, these extensions allows PHPStan to understand `symfony`, `phpunit`, `deprecation`, and `php parser`. without those extensions we had to ignore some errors. 

PHPStan baseline will be changed in this PR as some errors will be removed and others will be added, I will start an issue with the log of phpstan run, so we can remove those issues as we go 




#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Only code review is required
